backend/src/LibBackend/SqlCompiler.fs

```diff
@@ -261,10 +261,22 @@ let rec lambdaToSql
     let name = randomString 10
     $"(@{name})", [ name, Sql.double v ]
 
-  | EString (_, v) ->
-    typecheck $"string \"{v}\"" TStr expectedType
-    let name = randomString 10
-    $"(@{name})", [ name, Sql.string v ]
+  | EString (_, parts) ->
+    let strParts, vars =
+      parts
+      |> List.map (fun part ->
+        match part with
+        | StringText (s) ->
+          typecheck $"string \"{s}\"" TStr expectedType
+          let name = randomString 10
+          name, [ name, Sql.string s ]
+        | StringInterpolation e ->
+          let strPart, vars = lts TStr e
+          $"{strPart}", vars)
+      |> List.unzip
+    let strPart = strParts |> List.map (fun s -> $"(@{s})") |> String.concat ""
```

Comment:
This doesn't look like it works to me. I think adding tests will help find the issues in the code.

Reply:
They are getting concatenated, but I am unsure where the second `@` symbol is coming from in the result.
```
DEBUG: strPart ((@INQNEOOAHY)(@(@test_PGIDQSML))(@IAQMQCCDKD))
DEBUG: vars ([(INQNEOOAHY, String "R"); (test_PGIDQSML, String "ache"); (IAQMQCCDKD, String "l")])
```

This is the test I am using:
```
[test.queryOne interpolated ] with DB Person
(let _ = prepFriends null in
 let test = "ache" in
 let _ = Test.inspect test "test"
 (DB.queryOneWithKey_v3 Person (fun p -> p.name = $"R{test}l"))
 |> Option.map_v1 (fun v -> v.rachel.name)) = Just "Rachel"
```

Reply:
They are getting concatinated, but I'm not sure why the test isn't passing

```
DEBUG: strPart ((@YMDJOIJYXI)(@test_LYCVLLTX)(@HVIKGHEOWE))
DEBUG: vars ([(YMDJOIJYXI, String "R"); (test_LYCVLLTX, String "ache"); (HVIKGHEOWE, String "l")])
```

This the test I am using:
```
[test.queryOne interpolated - one] with DB Person
(let _ = prepFriends null in
 let test = "achel" in
 (DB.queryOneWithKey_v3 Person (fun p -> p.name = $"R{test}"))
 |> Option.map_v1 (fun v -> v.rachel.name)) = Just "Rachel"
```

Updated code:
```
  | EString (_, parts) ->
    let strParts, vars =
      parts
      |> List.map (fun part ->
        match part with
        | StringText (s) ->
          typecheck $"string \"{s}\"" TStr expectedType
          let name = randomString 10
          $"(@{name})", [ name, Sql.string s ]
        | StringInterpolation e ->
          let strPart, vars = lts TStr e
          strPart, vars)
      |> List.unzip
    let strPart = strParts |> String.concat ""
    let vars = vars |> List.concat
    strPart, vars
```

Reply:
If you put in a debug statement which prints out the full SQL, it will be possible to see what's wrong.

My sense is  that you need the postgres `concat` fn (see https://www.postgresql.org/docs/9.6/functions-string.html)

Reply:
Fixed, but I'm not sure if this is what you were looking for or if there's a better way to do it.